src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -700,19 +700,13 @@ public static function getHttpMethodParameterOverride()
     }
 
     /**
-     * Gets a "parameter" value.
+     * Gets a "parameter" value from any bag.
      *
-     * This method is mainly useful for libraries that want to provide some flexibility.
+     * This method is mainly useful for libraries that want to provide some flexibility. If you don't need the
+     * flexibility in controllers, it is better to explicitly get request parameters from the appropriate
+     * public property instead (attributes, query, request).
      *
-     * Order of precedence: GET, PATH, POST
-     *
-     * Avoid using this method in controllers:
-     *
-     *  * slow
```

Comment:
this is not relevant anymore with the removal of `deep` and also since the refactoring in #12369